app/Http/Controllers/ApiController/MainController.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace App\Http\Controllers\ApiController;
+
+use App\Http\Controllers\Controller;
+use App\User;
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Validator;
+
+
+class MainController extends Controller
+{
+    public function register(Request $request)
+    {
+        $validator = Validator::make($request->all(), [
+
+            'name' => 'required|string',
+            'email' => 'required|string|email|unique:users',
+            'password' => 'required|string|confirmed'
+        ]);
+
+        if ($validator->fails()) {
+            return response()->json(['error' => $validator->errors(), 'message' => 'validation error'], 422);
+        }
+
+        $user = new User();
+        $user->name = $request->name;
+        $user->email = $request->email;
+        $user->password = bcrypt($request->password);
+        $user->save();
+        // 201 status code used to show created data
+        return response()->json(['data' => $user, 'message' => 'User Created Successfully !..'], 201);
+    }
+
+    public function login(Request $request)
+    {
+        $validator = Validator::make($request->all(), [
+            'email' => 'required|string|email',
+            'remember_me' => 'boolean'
+        ]);
+
+        $userCredentials = request(['email', 'password']);
+        if (!Auth::attempt(['email' => $request->email, 'password' => $request->password])) {
+            return response()->json(['message' => 'Unauthrized'], 401);
+        }
+        $user = $request->user();
+        $tokenResult = $user->createToken('User Personal Acess Token');
+        $token = $tokenResult->token;
```

Comment:
You are authenticating the user using the `default` guard which is `web`.

Then creating a token using the authenticated user. 

You need to use the `api` guard here.